lib/pgfinder/pgio.py

```diff
@@ -171,6 +171,14 @@ def theo_masses_reader(file: Union[str, Path]) -> pd.DataFrame:
                 "columns. Have you checked the format of your database against one of the built-in databases?"
             )
         ) from e
+    # Check that all structures are followed by "|n" where n is one or more digits
+    if not theo_masses_df["Inferred structure"].str.contains(r"\|\d+$").all():
+        raise UserError(
```

Comment:
Codecov is saying this eventuality isn't tested.

Should be relatively straight-forward to construct a test for this (and other checks if they're not already covered) by having a simple CSV without the `|[1|2]` construct and using the...

```python
@pytest.mark.parameterize(
    "file,error", 
    [
     
   ("tests/resources/file1.csv", UserError),
     
   ("tests/resources/file2.csv", ValueError),
]
)
def test_exceptions(file: str, error) -> None:
    with pytest.raises(error)
        assert theo_masses_reader(file)
```

(or similar, bit of a rough hack/guess at the code).

Reply:
It's true that none of the exceptions are currently tested, I think, so I'll try to add in some proper tests in a bit!

Reply:
Thanks @TheLostLambda :+1:

Reply:
Sorry for the long silence @ns-rse ! How would you feel about writing up a Github issue for writing these tests (there are a lot of them and I might not have the time for a little while) and merging this PR so that the e2e tests are fixed for things like dependency bumping?

Reply:
(To clarify, I would write up the Github issue!)

Reply:
I'm of the opinion, based on my own failings in the past, that tests should be written when features are implemented. They purpose is after all to ensure that the code we write functions as expected and we ensure that by writing tests. In the past I've found that I let things slip or forget to update tests.

I've created the issue to remind us of this need in #251 mainly because I've started using the excellent browser extension [Refined Github](https://github.com/refined-github/refined-github/) that simplifies the GitHub interface and adds useful features, one of which is an additional item to the `...` of each message in PRs to _Reference in new issue_ which takes the text of the post and creates a new issue out of it, including a link back to the original discussion at the bottom. Brilliant extension with a huge amount of tweaks which I'm only just starting to familiarise myself with.